src/bun.js/webcore.zig

```diff
@@ -537,19 +477,59 @@ pub const Crypto = struct {
             if (options_value.getTruthy(globalThis, "maxmem")) |value| {
                 const p_int = value.to(i64);
                 if (p_int < 0 or !value.isNumber()) {
-                    return globalThis.createInvalidArgs("maxmem must be a positive integer", .{});
+                    return throwInvalidParams(
+                        globalThis,
+                        "RangeError",
+                        "Invalid scrypt params\n\n N must be a positive integer\n",
+                        .{},
+                    );
                 } else if (p_int != 0) {
                     maxmem = @intCast(usize, p_int);
                 }
             }
         }
 
-        if (cost < 2 or cost > 0x3fffffff) {
-            return globalThis.createRangeError("N must be greater than 1 and less than 2^30", .{});
+        if (blockSize == null) blockSize = 8;
+        if (cost == null) cost = 16384;
+        if (parallelization == null) parallelization = 1;
+
+        if (cost.? < 2 or cost.? > 0x3fffffff) {
+            return throwInvalidParams(
+                globalThis,
+                "RangeError",
+                "Invalid scrypt params\n\n N must be greater than 1 and less than 2^30\n",
+                .{},
+            );
         }
 
-        if (cost == 0 or (cost & (cost - 1)) != 0) {
-            return globalThis.createRangeError("N must be a power of 2 greater than 1", .{});
+        if (cost.? == 0 or (cost.? & (cost.? - 1)) != 0) {
+            return throwInvalidParams(
+                globalThis,
+                "RangeError",
+                "Invalid scrypt params\n\n N must be a power of 2 greater than 1\n",
+                .{},
+            );
+        }
+
+        if (keylen == 0) {
+            if ((BoringSSL.EVP_PBE_scrypt(
+                null,
+                0,
+                null,
+                0,
+                cost.?,
+                blockSize.?,
+                parallelization.?,
+                maxmem,
+                null,
+                0,
+            ) != 1)) {
+                return throwInvalidParams(globalThis, "RangeError", "Invalid scrypt params\n", .{});
+            }
+
+            var slice: []u8 = undefined;
+            slice.len = 0;
+            return JSC.ArrayBuffer.create(globalThis, slice, .ArrayBuffer);
```

Comment:
?

Reply:
oh

Can you create a special function for returning an empty ArrayBuffer? this looks really sketch / a mistake as-is

Reply:
👍